client/delegate-auth/delegate-auth-script.ts

```diff
@@ -0,0 +1,200 @@
+// In other words, we need to have an example application which does the following:
+
+// Creates an IAM (from iam-client-lib) instance representing an asset owner
+
+// Creates a new asset using iam-client-lib registerAsset()
+
+// Generates a new secp256k1 keypair
+
+// Adds the new key to the assets verification methods (see switchboard-dapp/verification.service.ts at 30aeaa15477709ef8a63504dcc707d81a20ce483 · energywebfoundation/switchboard-dapp  for example for how this is done in Switchboard)
+
+// Creates an IAM instance representing an asset using the private key from generate keypair
+
+// Creates an identity token (or uses token from iam initialization) and authenticates as the asset DID to the iam-client-examples backend
+
+// To start the nodejs “app” could just be a script that runs through the above steps and outputs logs describing what is happening. (It would be awesome if the logs had emojis   )
+
+import {
+    IAM,
+    Encoding,
+    PubKeyType,
+    Algorithms,
+    DIDAttribute,
+    setChainConfig,
+    WalletProvider,
+    setCacheClientOptions,
+} from 'iam-client-lib';
+import axios from 'axios';
+import * as emoji from 'node-emoji';
+import { Keys } from '@ew-did-registry/keys';
+import { utils, Signer, providers } from 'ethers';
+import { InitializeData } from 'iam-client-lib/dist/src/iam';
+
+
+type HeaderType = { alg?: string; typ?: string; };
+type SignerType = providers.JsonRpcSigner | Signer;
+type PayloadType = { iss: string; claimData: { blockNumber: string };}
+
+const backendUrl =  'http://localhost:3333';
+
+const connectIAM = async (privateKey: string) : Promise<{iamAgent : IAM, connectionInfos: InitializeData | undefined}> => {
+    // const privateKey = '14c4ce13e2ab410ac230f40a803bb2e978feaf6bd847bf0712087189d7493aa1';
+
+    let connectionInfos = undefined;
+
+        const iamAgent: IAM = new IAM({
+            rpcUrl: "https://volta-rpc-vkn5r5zx4ke71f9hcu0c.energyweb.org",
+            privateKey,
+        });
+        setCacheClientOptions(73799, {
+            url: "https://volta-identitycache.energyweb.org/",
+        });
+        
+        setChainConfig(73799, {
+            rpcUrl: "https://volta-rpc-vkn5r5zx4ke71f9hcu0c.energyweb.org"
+        });
+        connectionInfos = await iamAgent.initializeConnection({
+            walletProvider: WalletProvider.WalletConnect,
+            initCacheServer: false,
```

Comment:
Why `false` here?

Reply:
I delsabled this since I couldn't register assets when it is set to true. There may be an issue with the connection to cacheServer, and I didn't investigate for now.

Reply:
Would you have an idea of why this could give me an 400 error when I set initCacheServer to true please ?